src/java.base/share/classes/java/lang/String.java

```diff
@@ -2429,10 +2429,65 @@ public int indexOf(int ch) {
      *          character sequence represented by this object that is greater
      *          than or equal to {@code fromIndex}, or {@code -1}
      *          if the character does not occur.
+     *
+     * @apiNote
+     * An invocation of this method returns -1 when {@code fromIndex} happens
+     * to be too large. The result is thus indistinguishable from a genuine
```

Comment:
Adding an apiNote to the existing indexOf(int, int) is good but I think it will need a bit word smithing, e.g. "happens to be too large" is a bit too casual. I think I would start with saying the method returns -1 if fromFrom is negative or >= the string length, it does not throw an exception if called with an out of range index.

Reply:
Done.
(A negative `fromIndex` does not necessarily result in -1, though.)